File: services/api/app/models/otp.py
from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String, func

from app.core.database import Base


class OtpCode(Base):
    __tablename__ = "otp_codes"

    id = Column(Integer, primary_key=True, index=True)
    email = Column(String, index=True, nullable=False)
    purpose = Column(String, nullable=False)  # e.g., "signup"
    code_hash = Column(String, nullable=False)
    expires_at = Column(DateTime(timezone=True), nullable=False)
    attempts = Column(Integer, nullable=False, default=0)
    last_sent_at = Column(DateTime(timezone=True), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: rhserver/platform/models/Otp.js
module.exports = (sequelize, DataTypes) => {
    const Otp = sequelize.define("Otp", {
        email: { type: DataTypes.STRING, allowNull: true },
        phone: { type: DataTypes.STRING, allowNull: true },
        code_hash: { type: DataTypes.STRING, allowNull: false },
        purpose: { type: DataTypes.STRING, defaultValue: "login" }, // signup, login, reset
        expires_at: { type: DataTypes.DATE, allowNull: false },
        last_sent_at: { type: DataTypes.DATE, defaultValue: DataTypes.NOW }
    }, {
        tableName: 'otp_codes',
        timestamps: true,
        underscored: true
    });

    return Otp;
};
